### PR TITLE
=str code comment the Inlet/InPort dichotomy

### PR DESCRIPTION
**Pull request by rkuhn**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/Shape.scala [hunk 1]
```diff
@@ -6,10 +6,28 @@ package akka.stream
 import scala.collection.immutable
 import scala.collection.JavaConverters._
 
+/**
+ * An input port of a StreamLayout.Module. This type logically belongs
+ * into the impl package but must live here due to how `sealed` works.
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: can it be `private [akka]`?

Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: No, since we use it in public signatures, in particular in Flexi*.

Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: then it does not logically belong to the impl package, and I think this is more confusing than helpful for the user. Perhaps the reason should be in a non-scaladoc comment instead?

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: The comment is a bit confusing TBH - maybe explain why exactly, or make it `//` comments?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: INTERNAL API is what you are looking for :)

**Comment by drewhk** [comment]
LGTM after clearly marking the port classes as INTERNAL API

File: akka-stream/src/main/scala/akka/stream/Shape.scala (hunk 1 above)
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: but he just said that they are not internal

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: Ah, ok. But then the comment is wrong about belonging to impl.

Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: yes, that was my point with the question :-)

Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: Yes, TBH I also thought they were internal until I checked … thanks for the nudge, @patriknw! Will rework.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: Alternative is to really move it into impl, to the StreamLayout file, and mark it as INTERNAL API. "sealed" only works in scala anyway.

Comment (rkuhn) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: Well, we use it in public API, users will use this type, so it is not `INTERNAL API` in any sense.

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/Shape.scala]: maybe rename to Untyped then?

**Comment by drewhk** [comment]
ping

**Comment by drewhk** [comment]
ping ping

**Comment by rkuhn** [comment]
pong pong

**Comment by drewhk** [comment]
ready for merge?

**Comment by rkuhn** [comment]
No :-( but I don’t want to rename them, so I’ll need to make the comment actually fit reality—unless you really thing that UntypedInlet is better for everyone.

**Comment by drewhk** [comment]
I don't think there has been any issue on the ML so far about this, so the names can remain.

**Pull request merged**